test/partiql-tests-runner/src/main/kotlin/org/partiql/runner/ConformanceComparison.kt

```diff
@@ -11,91 +11,208 @@ fun main(args: Array<String>) {
+                    }
+                    this.appendLine("</details>")
+                } else {
+                    this.appendLine("${failureFirstPassingSecond.size} test(s) were previously failing but now pass. Before merging, confirm they are intended to pass")
+                }
+            }
+        }
+}
 
-Number failing in Base ($origCommitId) but now pass: ${failureOrigPassingNew.count()}
-"""
-    )
+class CrossEngineReportAnalyzer(private val first: Report, private val second: Report) : ReportAnalyzer(first, second) {
+    override val reportTitle: String = "Conformance comparison report-Cross Engine"
+    override fun generateComparisonReport() =
+        buildString {
+            this.appendLine(
+                """### $reportTitle
+| | Base ${first.engine} | ${second.engine} | +/- |
+| --- | ---: | ---: | ---: |
+| % Passing | ${"%.2f".format(firstPassingPercent)}% | ${"%.2f".format(secondPassingPercent)}% | ${"%.2f".format(secondPassingPercent - firstPassingPercent)}% |
+| :white_check_mark: Passing | $firstPassingSize | $secondPassingSize | ${secondPassingSize - firstPassingSize} |
+| :x: Failing | $firstFailingSize | $secondFailingSize | ${secondFailingSize - firstFailingSize} |
+| :large_orange_diamond: Ignored | $firstIgnoreSize | $secondIgnoreSize | ${secondIgnoreSize - firstIgnoreSize} |
+| Total Tests | $firstTotalSize | $secondTotalSize | ${secondTotalSize - firstTotalSize} |
+                """.trimIndent()
+            )
+            this.appendLine(
+                """
+Number passing in both: ${passingInBoth.count()}
 
-    if (!passingOrigFailingNew.isEmpty()) {
-        comparisonReportFile.appendText(
-            "\n:interrobang: CONFORMANCE REPORT REGRESSION DETECTED :interrobang:. The following test(s) were previously passing but now fail:\n<details><summary>Click here to see</summary>\n\n"
-        )
-        passingOrigFailingNew.forEach { testName ->
-            comparisonReportFile
-                .appendText("- ${testName}\n")
-        }
-        comparisonReportFile
-            .appendText("\n</details>")
-    }
+Number failing in both: ${failingInBoth.count()}
 
-    if (!failureOrigPassingNew.isEmpty()) {
-        comparisonReportFile.appendText(
-            "\nThe following test(s) were previously failing but now pass. Before merging, confirm they are intended to pass: \n<details><summary>Click here to see</summary>\n\n"
-        )
-        failureOrigPassingNew.forEach { testName ->
-            comparisonReportFile
-                .appendText("- ${testName}\n")
+Number passing in ${first.engine} engine but fail in ${second.engine} engine: ${passingFirstFailingSecond.count()}
+
+Number failing in ${first.engine} engine but pass in ${second.engine} engine: ${failureFirstPassingSecond.count()}
+                """.trimIndent()
+            )
+            if (passingFirstFailingSecond.isNotEmpty()) {
+                if (passingFirstFailingSecond.size < 10) {
```

Comment:
For cross-engine, could still be helpful to list all the tests passing engine first but not passing in second.